frc_characterization/newproject/__init__.py

```diff
@@ -29,6 +31,28 @@
 log_format = "%(asctime)s:%(msecs)03d %(levelname)-8s: %(name)-20s: %(message)s"
 
 
+class Tests(Enum):
+    ARM = "Arm"
+    ELEVATOR = "Elevator"
+    DRIVETRAIN = "Drivetrain"
+    SIMPLE = "Simple"
+
+
+class Units(Enum):
```

Comment:
Thoughts on this:
- Split into `RotationUnits` or `DistanceUnits`?
- Definitely remove `RotUnits` with the `VALDegrees`
 	- you could use `Units.DEGREES.name.title()` and then use the `DEGREES.value` to store the value rather than doing `RotUnits.VALDegrees`

I feel like it's weird to lump these all together... surely `pint` would be able to convert degrees -> rotations, etc and with the `isRotation` function you could determine whether to add the `ticksPerRotation` thing?

Reply:
Yeah I should probably utilize pint for this as well

Reply:
Would something like this be ideal for this task?
```
class Units(Enum):
    def to_label(self):
        text = self.name
        return text[0].upper() + text[1:].lower()
    
    DEGREES= some ureg
```

Then we can use the .to_label() to reference it as a string, and also use the ureg for unit conversions.

Reply:
Yup.

Note that python has the `.title()` (or `.capitalize()`, there's a slight difference I can't remember) that would be better than the manual string manipulation.
You may also want to do a `name.replace("_", " ")` as for things like `SIMPLE_MOTOR` it would appear nicer as `Simple Motor`. 

However saying this, the conversion from the exported label may not work, i.e. `Units(MechanismType.to_label())` probably wouldn't work

Reply:
Would it be better to just do this change for the Units and leave the tests set to their string counterparts?

Reply:
Seems kinda like abusing enums a bit to me.

I definitely don't like the `_all_values[1]` though; would definitely need to replace that with a `@property` or another attribute to refer to it.

Reply:
How about something like this?

```
class Units(Enum):

    __ureg__ = pint.UnitRegistry()
    
    DEGREES= "Degrees", __ureg__.degree
    
    def __new__(cls, *values):
        obj = object.__new__(cls)
        # first value is canonical value
        obj._value_ = values[0]
        obj.unit = values[1]
        return obj

    
Units("Degrees").unit # -> degree
Units("Degrees").value # -> "Degrees"



```

Reply:
Currently, I'm still using if statements, but let me know if you think that sticking the pint values on the enum would be a better option.

Reply:
Oops, I forgot to respond to this.

That looks fine to me, feel free to add it in place.

Reply:
[Integrated.](https://github.com/wpilibsuite/frc-characterization/blob/ccae082fde1ed5d7fee5194f9f66bb98c168fd9f/frc_characterization/newproject/__init__.py#L42-#L59)